applications/vanilla/views/modules/helper_functions.php

```diff
@@ -16,8 +16,12 @@ function writeModuleDiscussion($discussion, $px = 'Bookmark', $showPhotos = fals
    </span>
 
             <div class="Title"><?php
-                echo anchor(htmlspecialchars($discussion->Name), discussionUrl($discussion).($discussion->CountCommentWatch > 0 ? '#Item_'.$discussion->CountCommentWatch : ''), 'DiscussionLink');
-                ?></div>
+                echo anchor(
+                    Gdn::formatService()->renderHTML($discussion->Name, \Vanilla\Formatting\Formats\TextFormat::FORMAT_KEY),
```

Comment:
I'm pretty sure this is not supposed to be encoded at all. DiscussionModel::calculate() calls htmlspecialchars().

Reply:
Dropped running things through the format-to-text in favor of applying HTML sanitization via `HtmlSanitizer` (and ultimately htmLawed).